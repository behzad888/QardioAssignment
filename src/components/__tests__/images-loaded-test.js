//@flow
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ImagesLoaded from '../images-loaded/images-loaded';

describe('while images loaded test running in a browser environment', () => {
  let container, wrapper;

  beforeEach(() => {
    jest.resetModules();

    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    ReactDOM.render(null, container);
    document.body.removeChild(container);
    container = null;
  });

  it('should render correctly', () => {
    wrapper = renderer
      .create(
        <div>
          <ImagesLoaded>
            <img src="https://www.getqardio.com/wp-content/uploads/2019/05/19_14_General_Summer_WD_HP_EN.jpg" />
            <img src="https://www.getqardio.com/wp-content/uploads/2019/05/19_14_General_Summer_WD_HP_EN.jpg" />
            <img src="https://www.getqardio.com/wp-content/uploads/2019/05/19_14_General_Summer_WD_HP_EN.jpg" />
            <img src="https://www.getqardio.com/wp-content/uploads/2019/05/19_14_General_Summer_WD_HP_EN.jpg" />
            <img src="https://www.getqardio.com/wp-content/uploads/2019/05/19_14_General_Summer_WD_HP_EN.jpg" />
          </ImagesLoaded>
        </div>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
