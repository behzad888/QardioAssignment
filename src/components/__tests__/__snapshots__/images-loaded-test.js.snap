// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`while images loaded test running in a browser environment should render correctly 1`] = `
<div>
  <div>
    <img
      src="https://www.getqardio.com/wp-content/uploads/2019/05/19_14_General_Summer_WD_HP_EN.jpg"
    />
    <img
      src="https://www.getqardio.com/wp-content/uploads/2019/05/19_14_General_Summer_WD_HP_EN.jpg"
    />
    <img
      src="https://www.getqardio.com/wp-content/uploads/2019/05/19_14_General_Summer_WD_HP_EN.jpg"
    />
    <img
      src="https://www.getqardio.com/wp-content/uploads/2019/05/19_14_General_Summer_WD_HP_EN.jpg"
    />
    <img
      src="https://www.getqardio.com/wp-content/uploads/2019/05/19_14_General_Summer_WD_HP_EN.jpg"
    />
  </div>
</div>
`;
